### ISSUE TITLE
Query range with position function

### ISSUE THREAD

**Issue by farshidtz** [antchfx/jsonquery#3]
Hi,

Thank you for this package.

I am trying to query a range, but the result is wrong when using `and` operator:
```go
package main

import (
	"fmt"
	"github.com/antchfx/jsonquery"
	"strings"
)

func main() {
	s := `{
		"phoneNumbers": [
		  {
			"type"  : "mobile",
			"number": "0123-4567-0001"
		  },
		  {
			"type"  : "home",
			"number": "0123-4567-0002"
		  },
		  {
			"type"  : "office",
			"number": "0123-4567-0003"
		  },
		  {
			"type"  : "fax",
			"number": "0123-4567-0004"
		  }
		]
	}`
	doc, err := jsonquery.Parse(strings.NewReader(s))
	if err != nil {
		panic(err)
	}

	fmt.Println("print [2,end):")
	for _, n := range jsonquery.Find(doc, "phoneNumbers/*[ position()>=2 ]/number") {
		fmt.Println(n.InnerText())
	}

	fmt.Println("print [2,4):")
	for _, n := range jsonquery.Find(doc, "phoneNumbers/*[ position()>=2 and position()<4 ]/number") {
		fmt.Println(n.InnerText())
	}
	
	fmt.Println("print [2,4):")
	for _, n := range jsonquery.Find(doc, "phoneNumbers/*[ position()<4 and position()>=2 ]/number") {
		fmt.Println(n.InnerText())
	}

}
```
Output:
```
print [2,end):
0123-4567-0002
0123-4567-0003
0123-4567-0004
print [2,4):
0123-4567-0002
0123-4567-0003
0123-4567-0004
print [2,4):
```

[Playground](https://play.golang.org/p/Xz4RVUjnWCp)

It returns extra entries or nothing at all depending on how you order the operands.

Any idea?

**Comment by zhengchun** [comment]
I have tested, this bug is by `xpath` package.

**Comment by zhengchun** [comment]
fixed in v1.1.4.
